mParticle-Swrve/MPKitSwrve.m

```diff
@@ -0,0 +1,383 @@
+#import "MPKitSwrve.h"
+
+/* Import your header file here
+*/
+#if defined(__has_include) && __has_include(<SwrveSDK/SwrveSDK.h>)
+#import <SwrveSDK/SwrveSDK.h>
+#else
+#import "SwrveSDK.h"
+#endif
+
+@implementation MPKitSwrve
+/*
+    mParticle will supply a unique kit code for you. Please contact our team
+*/
++ (NSNumber *)kitCode {
+    return @216;
+}
+
++ (void)load {
+    MPKitRegister *kitRegister = [[MPKitRegister alloc] initWithName:@"Swrve" className:@"MPKitSwrve"];
+    [MParticle registerExtension:kitRegister];
+}
+
+- (MPKitExecStatus *)execStatus:(MPKitReturnCode)returnCode {
+    return [[MPKitExecStatus alloc] initWithSDKCode:self.class.kitCode returnCode:returnCode];
+}
+
+#pragma mark - MPKitInstanceProtocol methods
+
+#pragma mark Kit instance and lifecycle
+- (MPKitExecStatus *)didFinishLaunchingWithConfiguration:(NSDictionary *)configuration {
+    NSString *appId = configuration[@"app_id"];
+    NSString *apiKey = configuration[@"api_key"];
+    if (!apiKey || !appId) {
+        return [self execStatus:MPKitReturnCodeRequirementsNotMet];
+    }
+    
+    _configuration = configuration;
+    _started=NO;
+    
+    return [self execStatus:MPKitReturnCodeSuccess];
+}
+
+- (void)start {
+    _init_called=NO;
+    static dispatch_once_t swrvePredicate;
+    
+    dispatch_once(&swrvePredicate, ^{
+        /*
+         Start your SDK here. The configuration dictionary can be retrieved from self->_configuration
+         */
+        int appId = [self.configuration[@"app_id"] intValue];
+        NSString *apiKey = self.configuration[@"api_key"];
+        SwrveConfig* config = [[SwrveConfig alloc] init];
+        MPKitAPI *kitAPI = [[MPKitAPI alloc] init];
+        FilteredMParticleUser *currentUser = [kitAPI getCurrentUserWithKit:self];
+        //MPIdentityApi* currentId = [[MParticle sharedInstance] identity];
+        NSNumber *mpid = currentUser.userId;
+        config.pushResponseDelegate = [UIApplication sharedApplication];
+        config.pushEnabled = YES;
+        config.autoCollectDeviceToken = NO;
+        config.pushNotificationEvents = [[NSSet alloc] init];
+        config.appGroupIdentifier = [@"group." stringByAppendingString:[[NSBundle mainBundle] bundleIdentifier]];
+        self->_started=YES;
+        
+        config.userId = mpid.stringValue;
+        if ([config.userId isEqual:@"0"]){
+            return;
+        }
+        
+        //        _configuration = self->_configuration;
+        [SwrveSDK sharedInstanceWithAppID: appId
+                                   apiKey: apiKey
+                                   config: config];
+        self->_init_called=YES;
+        
+//        self->_started = YES;
+        
```

Comment:
Any special reason y r we calling just `self->_init_called=YES;` instead of `self->_started = YES;` like their example? mean we do init our SDK like the code at top of this we also start it.

Reply:
sorry @tiescoo, only saw this today. self->_started is set to YES on line 64. The reason I have separated these is because we need to return _started to indicate that they should pass messages to swrve, but we may not have called init, since we need the mpid to be returned (will only impact very first launch)